app/services/setting_registry.rb

```diff
@@ -52,9 +65,26 @@ def []=(name, value)
     definition = find(name)
     raise ::Foreman::Exception.new(N_("Setting definition for '%s' not found, can not set"), name) unless definition
     db_record = _find_or_new_db_record(name)
-    db_record.value = value
-    db_record.save!
-    definition.value = db_record.value
+    db_record.update!(value: value)
+  end
+
+  def set_user_value(name, value)
+    definition = find(name)
+    raise ActiveRecord::RecordNotFound.new(N_("Setting definition for '%s' not found, can not set") % name, Setting, name) unless definition
+    db_record = _find_or_new_db_record(name)
+    db_record.send(:set_setting_type_from_value)
+
+    type = value.class.to_s.downcase
+    type = 'boolean' if type == "trueclass" || type == "falseclass"
+    case type
+    when 'string'
+      db_record.parse_string_value(value)
+    when db_record.settings_type
+      db_record.value = value
+    else
+      raise ::Foreman::Exception.new(N_('expected a value of type %s'), @setting.settings_type)
```

Comment:
You don't need to perform the interpolation of `%s` here?

Reply:
yeah, this is correct, the interpolation is for recordnotfound execption, that doesn't do any translation for message.
So the Marek's change request was also wrong, getting it back.